File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/drivers/default_driver_impl/rat/tpm2d/TPM2dProverConfig.java
package de.fhg.aisec.ids.idscp2.drivers.default_driver_impl.rat.tpm2d;

import org.checkerframework.checker.nullness.qual.NonNull;

import java.security.cert.Certificate;

/**
 * A configuration class for TPM2d RatPriver driver
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
public class TPM2dProverConfig {

  private Certificate remoteCertificate;
  @NonNull private String tpm2dHost;

  private TPM2dProverConfig() {
    tpm2dHost = System.getenv("TPM_HOST") != null ?
        System.getenv("TPM_HOST") : "localhost";
  }

  public static class Builder {
    private static final TPM2dProverConfig config = new TPM2dProverConfig();

    @NonNull
    public Builder setRemoteCertificate(Certificate remoteCert) {
      config.remoteCertificate = remoteCert;
      return this;
    }

    @NonNull
    public Builder setTpmHost(String host) {
      config.tpm2dHost = host;
      return this;
    }

    public TPM2dProverConfig build() {
      return config;
    }
  }

  public Certificate getRemoteCertificate() {
    return remoteCertificate;
  }

  @NonNull
  public String getTpm2dHost() {
    return tpm2dHost;
  }
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/default_drivers/rat/tpm2d/TPM2dProverConfig.kt
package de.fhg.aisec.ids.idscp2.default_drivers.rat.tpm2d

import java.security.cert.Certificate

/**
 * A configuration class for TPM2d RatPriver driver
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
class TPM2dProverConfig private constructor() {
    var remoteCertificate: Certificate? = null
        private set
    var tpm2dHost: String
        private set

    class Builder {
        fun setRemoteCertificate(remoteCert: Certificate?): Builder {
            config.remoteCertificate = remoteCert
            return this
        }

        fun setTpmHost(host: String): Builder {
            config.tpm2dHost = host
            return this
        }

        fun build(): TPM2dProverConfig {
            return config
        }

        companion object {
            private val config = TPM2dProverConfig()
        }
    }

    init {
        tpm2dHost = if (System.getenv("TPM_HOST") != null) System.getenv("TPM_HOST") else "localhost"
    }
}